receiver/awscontainerinsightreceiver/internal/cadvisor/extractors/fs_extractor.go

```diff
@@ -80,11 +79,8 @@ func NewFileSystemMetricExtractor(logger *zap.Logger) *FileSystemMetricExtractor
 	fse := &FileSystemMetricExtractor{
 		logger: logger,
 	}
-	if p, err := regexp.Compile(allowList); err == nil {
-		fse.allowListRegexP = p
-	} else {
-		logger.Error("NewFileSystemMetricExtractor set regex failed", zap.Error(err))
-	}
+	p := regexp.MustCompile(allowList)
```

Comment:
This is not appropriate as it can result in a runtime panic.

Reply:
I think this line would not result in runtime panic, since `allowList` is a legal string const 

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/0d2a8fc7d4ed5a4adf6d9a14a5e3ced1602ee894/receiver/awscontainerinsightreceiver/internal/cadvisor/extractors/fs_extractor.go#L27-L29

So the MustCompile will panic only if someone change the `allowList`'s value in code.

Reply:
I didn't understand your point clearly. You mean I replace the allowList in function like following?
```
p := regexp.MustCompile(allowList)
```
to
```
p := regexp.MustCompile("^tmpfs$|^/dev/|^overlay$")
```

Reply:
No, I meant to replace

```go
 const ( 
 	allowList = "^tmpfs$|^/dev/|^overlay$" 
 ) 
```

with this:

```go
var allowedPaths = regexp.MustCompile(`^(tmpfs|\/dev\/.*|overlay)$`)
```

(rewriting the regex a bit to clarify and avoid potential character interpretation issues)

Then it doesn't even need to be a member on `fse` as it's a package global value without any particular state that is safe for concurrent use.